docs/architecture/adr-074-implicit-msg-signers.md

```diff
@@ -0,0 +1,126 @@
+addresses. How are we to accommodate this? Should there be a lookup into `x/auth` to check if an
+address belongs to a module or not? All of these thorny questions are delaying the delivery of
+inter-module message passing because we do not want an implementation that is overly complex.
+There are many use cases for inter-module message passing which are still relevant, the most
+immediate of which is a more robust denom management system in `x/bank` `v2` which is being explored
+in [ADR 071](https://github.com/cosmos/cosmos-sdk/pull/20316).
+
+## Alternatives
+
+Alternatives that have been considered are extending the current `x/tx` signer extraction system
+to inter-module message passing as defined in [ADR 033](./adr-033-protobuf-inter-module-comm.md).
+
+## Decision
+
+We have decided to introduce a new `MsgV2` standard whereby the signer of the message is implied
+by the credentials of the party sending it. These messages will be distinct from the existing messages
+and define new semantics with the understanding that signers are implicit.
+
+In the case of messages passed internally by a module or `x/account` instance, the signer of a message
+will simply be the main root address of the module or account sending the message. An interface for
+safely passing such messages to the message router will need to be defined.
+
+In the case of messages passed externally in transactions, `MsgV2` instances will need to be wrapped
+in a `MsgV2` envelope:
+```protobuf
+message MsgV2 {
+  string signer = 1;
+  google.protobuf.Any msg = 2;  
+}
+```
+
+Because the `cosmos.msg.v1.signer` annotation is required currently, `MsgV2` types should set the message option
+`cosmos.msg.v2.is_msg` to `true` instead.
+
+Here is an example comparing a v1 an v2 message:
+```protobuf
+// v1
+message MsgSendV1 {
+  string from_address = 1 [(cosmos.msg.v1.signer) = true];
+  string to_address = 2;
+  repeated Coin amount = 3;
+}
+
+// v2
+message MsgSendV2 {
+  option (cosmos.msg.v2.is_msg) = true;
+  // from address is implied by the signer
+  string to_address = 1;
+  repeated Coin amount = 2;
+}
+```
+
+Modules defining handlers for `MsgV2` instances will need to extract the sender from the `context.Context` that is
+passed in. An interface in `core` which will be present on the `appmodule.Environment` will be defined for this purpose:
+```go
+type GetSenderService interface {
+  GetSender(ctx sdk.Context) []byte
+}
+```
+
+Sender addresses that are returned by the service will be simple `[]byte` slices and any bech32 conversion will be
+done by the framework.
+
+## Consequences
+
+### Backwards Compatibility
+
+This design does not depreciate the existing method of embedded signers in `Msg`s and is totally compatible with it.
+
+### Positive
+
+* Allows for a simple inter-module communication design which can be used soon for the `bank` `v2` redesign.
+* Allows for simpler client implementations for messages in the future.
+
```

Comment:
the framework doesnt need to decode the entire message in order to get the signer now. We could do things in which we decode only the first field since we know the location every time